lacework/resource_lacework_alert_channel_datadog.go

```diff
@@ -111,80 +111,70 @@ func resourceLaceworkAlertChannelDatadogCreate(d *schema.ResourceData, meta inte
-	err = validateDatadogAlertChannelResponse(&response)
 	if err != nil {
 		return err
 	}
 
-	integration := response.Data[0]
+	integration := response.Data
 	d.SetId(integration.IntgGuid)
 	d.Set("name", integration.Name)
 	d.Set("intg_guid", integration.IntgGuid)
 	d.Set("enabled", integration.Enabled == 1)
 	d.Set("created_or_updated_time", integration.CreatedOrUpdatedTime)
 	d.Set("created_or_updated_by", integration.CreatedOrUpdatedBy)
-	d.Set("type_name", integration.TypeName)
+	d.Set("type_name", integration.Type)
 	d.Set("org_level", integration.IsOrg == 1)
 
 	if d.Get("test_integration").(bool) {
-		log.Printf("[INFO] Testing %s integration for guid %s\n", api.DatadogChannelIntegration, d.Id())
+		log.Printf("[INFO] Testing %s integration for guid %s\n", api.DatadogAlertChannelType, d.Id())
 		if err := VerifyAlertChannelAndRollback(d.Id(), lacework); err != nil {
 			return err
 		}
-		log.Printf("[INFO] Tested %s integration with guid %s successfully\n", api.DatadogChannelIntegration, d.Id())
+		log.Printf("[INFO] Tested %s integration with guid %s successfully\n", api.DatadogAlertChannelType, d.Id())
 	}
 
-	log.Printf("[INFO] Created %s integration with guid: %s\n", api.DatadogChannelIntegration, d.Id())
+	log.Printf("[INFO] Created %s integration with guid: %s\n", api.DatadogAlertChannelType, d.Id())
 	return nil
 }
 
 func resourceLaceworkAlertChannelDatadogRead(d *schema.ResourceData, meta interface{}) error {
 	lacework := meta.(*api.Client)
 
-	log.Printf("[INFO] Reading %s integration with guid %s\n", api.DatadogChannelIntegration, d.Id())
-	response, err := lacework.Integrations.GetDatadogAlertChannel(d.Id())
+	log.Printf("[INFO] Reading %s integration with guid %s\n", api.DatadogAlertChannelType, d.Id())
+	response, err := lacework.V2.AlertChannels.GetDatadog(d.Id())
 	if err != nil {
 		return err
 	}
 
-	for _, integration := range response.Data {
-		if integration.IntgGuid == d.Id() {
-			d.Set("name", integration.Name)
-			d.Set("intg_guid", integration.IntgGuid)
-			d.Set("enabled", integration.Enabled == 1)
-			d.Set("created_or_updated_time", integration.CreatedOrUpdatedTime)
-			d.Set("created_or_updated_by", integration.CreatedOrUpdatedBy)
-			d.Set("type_name", integration.TypeName)
-			d.Set("org_level", integration.IsOrg == 1)
-			d.Set("datadog_site", integration.Data.DatadogSite)
-			d.Set("datadog_service", integration.Data.DatadogService)
-
-			log.Printf("[INFO] Read %s integration with guid %s\n",
-				api.DatadogChannelIntegration, integration.IntgGuid)
-			return nil
-		}
-	}
-
-	d.SetId("")
+	d.Set("name", response.Data.Name)
+	d.Set("intg_guid", response.Data.IntgGuid)
+	d.Set("enabled", response.Data.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.CreatedOrUpdatedBy)
+	d.Set("type_name", response.Data.Type)
+	d.Set("org_level", response.Data.IsOrg == 1)
+	d.Set("api_key", response.Data.Data.ApiKey)
```

Comment:
The API server will never return this key, we should not add this Set statement. (which it wasn't here before)
```suggestion
```

Reply:
Definitely a blocker 👆🏽